foundations/src/telemetry/driver.rs

```diff
@@ -0,0 +1,112 @@
+use crate::utils::feature_use;
+use crate::BootstrapResult;
+use futures_util::future::BoxFuture;
+use futures_util::stream::{FuturesUnordered, Stream};
+use futures_util::FutureExt;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+feature_use!(cfg(feature = "telemetry-server"), {
+    use super::server::TelemetryServerFuture;
+    use anyhow::anyhow;
+    use hyper::Server;
+    use std::net::SocketAddr;
+});
+
+/// A future that drives async telemetry functionality and that is returned
+/// by [`crate::telemetry::init`].
+///
+/// You would usually need to pass it to [`tokio::spawn`] once all the [security syscall-related]
+/// configuration steps are completed.
+///
+/// [security syscall-related]: `crate::security`
+pub struct TelemetryDriver {
+    #[cfg(feature = "telemetry-server")]
+    server_addr: Option<SocketAddr>,
+
+    #[cfg(feature = "telemetry-server")]
+    server_fut: Option<TelemetryServerFuture>,
+
+    tele_futures: FuturesUnordered<BoxFuture<'static, BootstrapResult<()>>>,
```

Comment:
if this is always going to have one future, why is this different from `server_fut`

Reply:
No, in the upcoming PRs it will be also used to drive exporters.